### ISSUE TITLE
asv fails to run on windows when pkg-config and OpenBLAS are set up properly

### ISSUE THREAD

**Issue by mattip** [numpy/numpy#24251]
running either 
```
python runtests.py --bench-compare <commit1> <commit2>
```

or 
```
spin bench --compare <commit1> <commit2>
```

will end up calling 
```
cd benchmark; asv continuous --factor 1.05 --bench main <commit1> <commit2>
```
Then asv creates a virtualenv for each commit, and uses a vanilla `pip install` to build NumPy. While this can work on macOS and linux (and use the system OpenBLAS), it will fail to correctly use OpenBLAS on windows, since the build also needs to
- copy the OpenBLAS DLL into the correct place
- build the `_distributor_init.py` file. In CI this is done via additional post-install code:

Is there a way to, if OpenBLAS is found, to do this as part of the build?

Here is what the various CI invocations to do these steps look like: 
https://github.com/numpy/numpy/blob/374e73d0690d5caa6dca45af8b181b0472594498/.github/workflows/windows_meson.yml#L63-L72
https://github.com/numpy/numpy/blob/374e73d0690d5caa6dca45af8b181b0472594498/.github/workflows/windows_clangcl.yml#L66-L75

(on azure this also shows the complicated command line needed)
https://github.com/numpy/numpy/blob/374e73d0690d5caa6dca45af8b181b0472594498/azure-steps-windows.yml#L38-L48

https://github.com/numpy/numpy/blob/374e73d0690d5caa6dca45af8b181b0472594498/tools/wheels/cibw_before_build.sh#L39-L56

**Comment by rgommers** [comment]
I think this is difficult to make work in any reasonable way. Basically, the poor design of Windows and the poor design of Python packaging tooling intersect here - `pip install numpy --no-binary` on Windows is also fundamentally broken, for the same reason.

One hack could be to let `asv` run `delvewheel` only on Windows.

**Comment by mattip** [comment]
I assume we could get meson to build the `_distributor_init.py` file built as part of the build scripts, conditioned on propogating some OpenBLAS config value. Could we get the DLL copied as part of the install step if we conditionally declare it a data file?

**Comment by mattip** [comment]
Or maybe this all should wait for OpenBLAS to become a wheel, which would basically solve the whole problem.

**Comment by rgommers** [comment]
> Or maybe this all should wait for OpenBLAS to become a wheel, which would basically solve the whole problem.

Yes, good point. +1 for waiting. It's not like this is a new problem, I don't think this ever worked, the `numpy.distutils` build has the same problems.
